relay-server/src/services/processor/playstation.rs

```diff
@@ -2,42 +2,328 @@
+
+    if let Some(app_info) = &prospero.app_info {
+        add_tag!("titleId".into(), app_info.title_id.into());
+
+        contexts.get_or_default::<AppContext>().app_version =
+            Annotated::new(app_info.version.into());
+    }
+
+    contexts.add(device_context);
+    contexts.add(os_context);
+    contexts.add(runtime_context);
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::borrow::Cow;
+    use std::collections::BTreeMap;
+
+    use relay_protocol::Value;
+
+    #[test]
+    fn test_event_release_setting() {
+        let prospero = ProsperoDump {
+            userdata: BTreeMap::from([("release", Cow::Borrowed("1.2.3.4"))]),
+            ..Default::default()
+        };
+
+        let mut event = Event::default();
+        update_sentry_event(&mut event, &prospero);
+        assert_eq!(
+            event.release,
+            Annotated::new(LenientString("1.2.3.4".into()))
+        );
+    }
+
+    #[test]
+    fn test_event_field_mapping() {
+        let prospero = ProsperoDump {
+            userdata: BTreeMap::from([
+                ("sentry.release", Cow::Borrowed("1.2.3.4")),
+                ("sentry.environment", Cow::Borrowed("production")),
+                ("sentry.user.username", Cow::Borrowed("janedoe")),
+                ("sentry.user.email", Cow::Borrowed("janedoe@example.com")),
+                ("santry.tag", Cow::Borrowed("other_value")),
+                ("other_tag", Cow::Borrowed("other_value")),
+            ]),
+            ..Default::default()
+        };
+
+        let mut event = Event::default();
+        update_sentry_event(&mut event, &prospero);
+
+        // Check the event fields
+        assert_eq!(
+            event.release,
+            Annotated::new(LenientString("1.2.3.4".into()))
+        );
+        assert_eq!(event.environment, Annotated::new("production".into()));
+
+        if let Some(user) = event.user.0 {
+            assert_eq!(
+                user.username,
+                Annotated::new(LenientString("janedoe".into()))
+            );
+            assert_eq!(user.email, Annotated::new("janedoe@example.com".into()));
+        } else {
+            panic!("User information not set in the event");
+        }
+
+        // Checks the tags are present
+        let tags = event.tags.value().unwrap();
+        assert_eq!(tags.get("release"), Some("1.2.3.4"));
+        assert_eq!(tags.get("environment"), Some("production"));
+        assert_eq!(tags.get("user.username"), Some("janedoe"));
+        assert_eq!(tags.get("user.email"), Some("janedoe@example.com"));
+        assert_eq!(tags.get("santry.tag"), Some("other_value"));
+        assert_eq!(tags.get("other_tag"), Some("other_value"));
+
+        // Check that the raw tags are not present
+        assert!(!tags.0.contains("sentry.release"));
+        assert!(!tags.0.contains("sentry.environment"));
+        assert!(!tags.0.contains("sentry.user.username"));
+        assert!(!tags.0.contains("sentry.user.email"));
```

Comment:
When you assert `tags` this is much easier and covers also cases you didn't think of (e.g. a new tag `foobar`).